Ref re show open tables

amorynan opened the pull request:
## What type of PR is this：
- [ ] bug
- [ ] feature
- [ ] enhancement
- [x] refactor
- [ ] others

## Which issues of this PR fixes ：
<!--
Usage: `Fixes #<issue number>`, or `Fixes (paste link of issue)`.
_If PR is about `failing-tests or flakes`, please post the related issues/tests in a comment and do not use `Fixes`_*
-->
Fixes #

## Problem Summary(Required) ：
<!-- (Please describe the changes you have made. In which scenarios will this bug be triggered and what measures have you taken to fix the bug?) -->

ZiheLiu committed:
[Bugfix] Cancel query when throwing any exception (#9005)

Commit:
update for ShowOpenTables

amorynan committed:
Merge branch 'StarRocks:main' into main

nshangyiming committed:
[Enhancement] support dynamically change slot per path conf (#8987)

stdpain committed:
[Bug] fix call JNI function in bthread in UDAF when load class failed (#8970)

when we provide a invalid UDAF function and failed in prepare stage,
~JavaGlobalRef() will be called in FragmentExecutor::_fail_cleanup,
which is executed in a bthread.

sduzh committed:
[Enhancement] Simple compaction scheduler for lake tables (#8850)

xlwh committed:
[Refactor] Remove old parser and analyzer code for database-related statements (#9028)

dulong41 committed:
Update sonar4fe.yml (#9027)

stdpain committed:
[Bugfix] Fix the problem that dictionary optimization cannot get the dictionary after exceeding the cache capacity (#8946)

Our dictionary fetching process is to determine if the dictionary exists first, and then to fetch the dictionary.
If the size of the dictionary cache exceeds the capacity under a certain moment,
then it will cause a random elimination.
This may lead to fetching empty dictionaries later

dirtysalt committed:
[Enhancement] avoid same table inner join when use heuristic method in left deep join (#9031)

cbcbq committed:
[Feature]add support for be graceful exit (#8726)

change graceful exit behaviour to:
1. BE tell FE it will exit soon
2. FE won't dispatch new fragments to BE exiting
3. BE loop wait all fragments done (maximum loop count is configurable), then stop servers and exit

liukun4515 committed:
[SR SQL Planner Summer Camp] Show char set planner (#8922)

murphyatwork committed:
[Feature] turn off the enable_shared_scan by default (#9015)

meegoo committed:
[Enhancement] Clarify stream load commit rpc error message (#8925)

HangyuanLiu committed:
Fix analyze status state replay bug (#8959)

kevincai committed:
[bugfix] Fix metric unit typo (#9045)

Signed-off-by: Kevin Xiaohua Cai <caixiaohua@starrocks.com>

HangyuanLiu committed:
Fix password parse string bug (#9038)

amorynan committed:
Merge branch 'StarRocks:main' into main

blackstar-baba committed:
[Bugfix] Fix mv version map issues (#8799)

Fix the problem that the materialized view cannot add partitions due to base table replace partition
Fix the problem that the version map was written to an incorrect version (wait for the insert overwrite to be completed and then write the version, and add a lock to ensure the correctness of the write)
Fix partition name relationships add and remove issues

amorynan committed:
Merge branch 'StarRocks:main' into main

hiliuxg committed:
[SR SQL Planner Summer Camp][Feature] alter table statement support of add list partition (#9019)

stdpain committed:
[Bugfix] fix "unsupport decode_dict_codes" error in late_materized (#9046)

After PR #8869, GlobalDictCodeColumnIterator won't support `decode_dict_codes(const int32_t* codes, size_t size, vectorized::Column* words)`,
we need to decode_dict_codes(const vectorized::Column& codes, vectorized::Column* words) instead it

stdpain committed:
[Bugfix] Fix DictMappingExpr rewrite multi times (#8950)

Because Expr and ExprContext are shared in OlapScanOperator.
And OlapScanPrepareOperator::pull_chunk() will be executed concurrently,
which can cause the rewrite function to be executed multiple times.
We should use std::call_once to avoid multiple rewrites

dulong41 committed:
Update add-approve-label.yml (#9064)

add workflow run_id to get the most recent pr information

amorynan committed:
Revert "update for ShowOpenTables"

This reverts commit 674dabf6bdde5f7f6c77dcba385369330f2b0d86.

amorynan committed:
Revert "update for ShowOpenTables"

This reverts commit 674dabf6bdde5f7f6c77dcba385369330f2b0d86.

amorynan committed:
Merge branch 'main' of github.com:amorynan/starrocks

amorynan committed:
Merge branch 'StarRocks:main' into main

blackstar-baba committed:
mv map persistence (#8879)

liuyehcf committed:
[Enhancement] Add invisible session variable 'profile_timeout' (#8999)

* [Enhancement] increase profile's waiting timeout

* set timeout to 1 minute

* update comments

* add invisiable session variable

* naming

HangyuanLiu committed:
[Enhancement] Optimized statistics scheduling and not collecting empty tables and partitions (#8958)

Smith-Cruise committed:
Fix unsable Slack links for new users. (#9067)

If you are a new Slack user, the original URL is invalid.

HangyuanLiu committed:
[Enhancement] Refactored statistics expired logic and support Drop statistics statement (#8960)

choury committed:
Skip create agent server when start cn (#9061)

Pslydhh committed:
Add compress debug info when disable minidump (#9043)

Compress debug info to reduce file size(1.7GB) to 600MB.

xlwh committed:
[Refactor] Remove old parser for partition & function related statements (#9083)

dulong41 committed:
[Actions] fix pr and run_id cannot use together (#9089)

* Update add-approve-label.yml

* Update add-approve-label.yml

* Update add-approve-label.yml

ZiheLiu committed:
[Enhancement] Use object instead of pointer for HyperLogLog::_hash_set (#9073)

#6885 uses pointer instead of object for HyperLogLog::_hash_set to reduce memory usage of HLL and not include phmap.h in hll.h.

However, as for reading HLL column from disk,

using pointer will allocate memory 4096 times for HLL hash_set of 4096 rows in a chunk,
while using object only need allocate 1 times for a chunk by _pool.reserve.

amorynan committed:
Merge branch 'StarRocks:main' into main

amorynan committed:
change name

amorynan committed:
change name

zombee0 committed:
Remove unused InfoSchemaDb ctor (#9093)

zombee0 committed:
[Refactor] add default_cluster in LocalMetaData and Database so that can remove it from stmt safely (#9088)

add default_cluster in LocalMetaData and Database so that can remove it from stmt safely.

CLAassistant commented:
[![CLA assistant check](https://cla-assistant.io/pull/badge/not_signed)](https://cla-assistant.io/StarRocks/starrocks?pullRequest=9097) <br/>Thank you for your submission! We really appreciate it. Like many open source projects, we ask that you all sign our [Contributor License Agreement](https://cla-assistant.io/StarRocks/starrocks?pullRequest=9097) before we can accept your contribution.<br/>**1** out of **2** committers have signed the CLA.<br/><br/>:white_check_mark: amorynan<br/>:x: amory<br/><hr/>**amory** seems not to be a GitHub user. You need a GitHub account to be able to sign the CLA. If you have already a GitHub account, please [add the email address used for this commit to your account](https://help.github.com/articles/why-are-my-commits-linked-to-the-wrong-user/#commits-are-not-linked-to-any-user).<br/><sub>You have signed the CLA already but the status is still pending? Let us [recheck](https://cla-assistant.io/check/StarRocks/starrocks?pullRequest=9097) it.</sub>

imay commented:
@amorynan you should make your commit email the same as github email.
there are two amorynan amory users. you should make all the commits submit by amorynan. then the CLA check will be OK.

femiiii committed:
[Doc]Update sql statement (#9092)

wuleistarrocks committed:
Fill cache configuration before creating new cache filesystem (#9087)

* Fill cache configuration before creating new cache filesystem
if local disk cache enabled

* Disable local disk cache if cache_dir set to empty

HangyuanLiu committed:
Fix join precedence error (#9063)

The precedence of the comma operator is less than that of INNER JOIN, CROSS JOIN, LEFT JOIN, and so on
In consecutive joins without on conditions, a left-deep tree should be generated, and if there is an on predicate in the last join, it should correspond to the join formed by all preceding tables and the join generated by the last table.

Pslydhh committed:
[Enhancement] support BIGINT/INT argument for window_funnel (#9032)

Reduce copy when access array daum.
Add BIGINT and INT as argument of window_funnel to remove cast costs.

amorynan committed:
update email and username for cla

amorynan committed:
update email and username for cla

wanpengfei-git commented:
#### [FE PR Coverage Check]

:heart_eyes: **pass** : 33 / 35 (94.29%)




#### file detail

|   |path|covered_line|new_line|coverage|not_covered_line_detail|
|----|----|----|----|----|----|
|:large_blue_circle:|com/starrocks/sql/parser/AstBuilder.java|9|11|81.82%|[951, 952]|
|:large_blue_circle:|com/starrocks/sql/ast/AstVisitor.java|1|1|100.00%|[]|
|:large_blue_circle:|com/starrocks/analysis/ShowOpenTableStmt.java|23|23|100.00%|[]|

amorynan committed:
update for ShowOpenTables

Seaven committed:
[Enhancement] CTE prune shuffle column (#8948)

amorynan committed:
Merge branch 'ref-re-showOpenTables' of github.com:amorynan/starrocks into ref-re-showOpenTables

murphyatwork committed:
Fix the thread-safety of phmap (#9099)

- `parallel_flat_hash_map::lazy_emplace` returns a iterator, which is not safe to read cause following insert would invalid it
- `parallel_flat_hash_map::lazy_emplace_l` 
  - It is safe to read if exists, or insert a value if not exists
  - This function would exclusively lock the inner map during the lookup or insert

abc982627271 committed:
Fix truncate lake table bug (#9002)

After truncated , showing partition and tablet would return 'unknown error', this pr fix this bug

Youngwb committed:
[BugFix] Fix CompoundPredicateOperator not equals when same children with different order (#8810)

sduzh committed:
[Enhancement] lake tablet GC (#9074)

amorynan commented:
> @amorynan you should make your commit email the same as github email. there are two amorynan amory users. you should make all the commits submit by amorynan. then the CLA check will be OK.

I try a new branch to solve this
